### PR TITLE
Ukhpqixe

### PR DESCRIPTION
**Pull request by mariana-furyk**

### CODE REVIEW

File: src/components/Artifacts/Artifacts.js [hunk 1]
```diff
@@ -99,25 +105,20 @@ const Artifacts = ({ artifactsStore, fetchArtifacts, match, history }) => {
...
-        tableHeaders={artifactsData.tableHeaders}
         yamlContent={artifactsStore.artifacts}
       />
     </>
   )
 }
 
 Artifacts.propTypes = {
-  artifactsStore: PropTypes.shape({}).isRequired,
-  fetchArtifacts: PropTypes.func.isRequired,
```
Comment (mavdryk) [inline src/components/Artifacts/Artifacts.js:120]: why did you delete these propTypes?

File: src/components/FunctionsPage/Functions.js [hunk 2]
```diff
@@ -98,30 +104,25 @@ const Functions = ({ fetchFunctions, functionsStore, match, history }) => {
...
         yamlContent={functionsStore.functions}
       />
     </>
   )
 }
 
 Functions.propTypes = {
-  fetchFunctions: PropTypes.func.isRequired,
   history: PropTypes.shape({}).isRequired,
-  functionsStore: PropTypes.shape({}).isRequired,
```
Comment (mavdryk) [inline src/components/FunctionsPage/Functions.js:124]: why did you delete these propTypes

File: src/components/JobsPage/Jobs.js [hunk 3]
```diff
@@ -3,18 +3,31 @@ import { connect } from 'react-redux'
...
 import Content from '../../layout/Content/Content'
 import Loader from '../../common/Loader/Loader'
 
-const Jobs = ({ fetchJobs, jobsStore, match, history }) => {
+const Jobs = ({
+  fetchJobs,
+  fetchWorkflows,
+  jobsStore,
+  match,
+  history,
```
Comment (mavdryk) [inline src/components/JobsPage/Jobs.js]: sort proprs

File: src/components/Artifacts/Artifacts.js [hunk 4]
```diff
@@ -15,6 +15,12 @@ import './artifacts.scss'
 const Artifacts = ({ artifactsStore, fetchArtifacts, match, history }) => {
```
Comment (mavdryk) [inline src/components/Artifacts/Artifacts.js]: sort props

File: src/components/Table/Table.js [hunk 5]
```diff
@@ -80,19 +82,21 @@ const Table = ({
     <>
       <TableView
         content={content}
-        detailsMenu={detailsMenu}
         groupFilter={groupFilter}
-        groupLatestItem={groupLatestItem}
+        groupLatestItem={
+          groupLatestItem.length ? groupLatestItem : groupWorkflowItems
```
Comment (mavdryk) [inline src/components/Table/Table.js]: use lodash.isEmpty

File: src/components/Table/Table.js [hunk 6]
```diff
@@ -6,72 +6,74 @@ import TableView from './TableView'
...
+        pageData.page,
+        generatedTableContent
+      )
 
       setGroupLatestItem(groupLatest)
+      setGroupWorkflowItems([])
+    } else if (groupFilter === 'workflow') {
+      const groupWorkflowItem = Object.keys(
+        groupedByWorkflow
+      )?.map(workflowId =>
```
Comment (mavdryk) [inline src/components/Table/Table.js]: you don't need `Object.keys` , you can use just map

Comment (mavdryk) [inline src/components/Table/Table.js]: you don't need `Object.keys` here, you can use just `map`

File: src/components/Table/table.scss [hunk 7]
```diff
@@ -9,112 +9,146 @@
...
       position: relative;
       display: flex;
-      min-height: 50px;
       flex-direction: row;
+      min-height: 50px;
+
       &.parent-row {
-        &:not(.parent-row-expanded) {
+
+        &:not(.parent-row-expanded):not( .isGroupedByWorkflow) {
```
Comment (mavdryk) [inline src/components/Table/table.scss]: don't use `isGroupedByWorkflow` class. It's too specific name for common component

File: src/components/Table/table.scss [hunk 8]
```diff
@@ -9,112 +9,146 @@
...
+        @include tableColumnFlex(2, 80px);
+      }
     }
-    .artifacts_popout,
-    .artifacts_download {
-      @include tableColumnFlex(0.5, 80px);
-      justify-content: center;
+
+    .artifacts {
+
```
Comment (mavdryk) [inline src/components/Table/table.scss]: delete empty row

File: src/components/Table/table.scss [hunk 9]
```diff
@@ -9,112 +9,146 @@
...
-    }
-    .artifacts_kind {
-      padding-left: 20px;
-      @include tableColumnFlex(0.5, 80px);
-    }
-    .artifacts_owner {
-      @include tableColumnFlex(0.5, 80px);
+
+    .jobs {
+
```
Comment (mavdryk) [inline src/components/Table/table.scss]: delete empty row

File: src/components/Table/table.scss [hunk 10]
```diff
@@ -9,112 +9,146 @@
...
+      &_popout,
+      &_download {
+        @include tableColumnFlex(0.5, 80px);
+        justify-content: center;
+      }
     }
+
     .functions {
+
```
Comment (mavdryk) [inline src/components/Table/table.scss]: delete empty row

File: src/components/Table/table.scss [hunk 11]
```diff
@@ -9,112 +9,146 @@
...
     &__row {
       position: relative;
       display: flex;
-      min-height: 50px;
       flex-direction: row;
+      min-height: 50px;
+
       &.parent-row {
-        &:not(.parent-row-expanded) {
+
```
Comment (mavdryk) [inline src/components/Table/table.scss]: delete empty row

File: src/components/Table/table.scss [hunk 12]
```diff
@@ -9,112 +9,146 @@
...
+        &:not(.parent-row-expanded):not( .isGroupedByWorkflow) {
           &:hover {
             background-color: $alabaster;
+
             .action_cell {
+
```
Comment (mavdryk) [inline src/components/Table/table.scss]: delete empty row

File: src/components/Table/table.scss [hunk 13]
```diff
@@ -9,112 +9,146 @@
...
         &-expanded {
           flex-direction: column;
         }
+
         .row_grouped-by {
           width: 100%;
+
           .table-body__row {
+
```
Comment (mavdryk) [inline src/components/Table/table.scss]: delete empty row

File: src/elements/JobsTableRow/JobsTableRow.js [hunk 14]
```diff
@@ -97,27 +99,33 @@ const JobsTableRow = ({
...
       ) : (
         <>
-          {Object.values(rowItem).map((value, i, arr) => {
+          {Object.values(rowItem).map((rowItemProp, i) => {
+            const currentItem = isGroupedByWorkflow
+              ? workflows.find(workflow => workflow.id === rowItem.uid.value)
+              : content.find(
+                  contentItemObj => contentItemObj.uid === rowItem.uid.value
+                )
+
```
Comment (mavdryk) [inline src/elements/JobsTableRow/JobsTableRow.js]: Again, you don't need `Object.values`

**Review by mavdryk — CHANGES_REQUESTED** [review]

File: src/components/Artifacts/Artifacts.js (hunk 1 above)
Comment (mariana-furyk) [reply src/components/Artifacts/Artifacts.js:120]: they come from store, we don't pass them through props

File: src/components/Table/table.scss [hunk 15]
```diff
@@ -9,112 +9,144 @@
...
               }
             }
           }
         }
+
+        &:not(.parent-row-expanded) {
+          &:hover {
+            background-color: $alabaster;
+          }
+        }
```
Comment (mavdryk) [inline src/components/Table/table.scss]:
```
&:not(.parent-row-expanded) {
          &:hover {
            background-color: $alabaster;
          }

          &:not(.parent-row_without-actions) {
            &:hover {
              .action_cell {
                .table-actions-container {
                  display: block;
                }
              }
            }
          }
        }
```

File: src/elements/JobsTableRow/JobsTableRow.js [hunk 16]
```diff
@@ -63,10 +82,17 @@ const JobsTableRow = ({
                   {Object.values(job).map((value, i) => {
                     const currentItem =
                       content.length > 0 &&
-                      content.find(item => item.uid === job.uid.value)
+                      content.find(
+                        contentItemObj => contentItemObj.uid === job.uid.value
+                      )
```
Comment (mavdryk) [inline src/elements/JobsTableRow/JobsTableRow.js]:
```suggestion
                      _.find(content, ['uid', job.uid.value])
```

File: src/elements/JobsTableRow/JobsTableRow.js [hunk 17]
```diff
@@ -63,10 +82,17 @@ const JobsTableRow = ({
                   {Object.values(job).map((value, i) => {
```
Comment (mavdryk) [inline src/elements/JobsTableRow/JobsTableRow.js]: you can get rid of `Object.values`

Comment (mavdryk) [reply src/elements/JobsTableRow/JobsTableRow.js]: and rename `value` and `i` variables

File: src/elements/JobsTableRow/JobsTableRow.js [hunk 18]
```diff
@@ -97,27 +123,33 @@ const JobsTableRow = ({
         </div>
       ) : (
         <>
-          {Object.values(rowItem).map((value, i, arr) => {
+          {map(rowItem, (rowItemProp, rowItemKey) => {
+            const currentItem = isGroupedByWorkflow
+              ? workflows.find(workflow => workflow.id === rowItem.uid.value)
+              : content.find(
+                  contentItemObj => contentItemObj.uid === rowItem.uid.value
+                )
```
Comment (mavdryk) [inline src/elements/JobsTableRow/JobsTableRow.js]: use _.find

File: src/elements/JobsTableRow/JobsTableRow.js [hunk 19]
```diff
@@ -97,27 +123,33 @@ const JobsTableRow = ({
...
+                key={new Date().getTime() + rowItemKey}
                 link={
-                  i === 0 &&
-                  `/projects/${match.params.projectName}/jobs/${content.length >
-                    0 &&
-                    content.filter(item => item.uid === rowItem.uid.value)[0]
-                      ?.uid}${
+                  rowItemKey === 'name' &&
+                  `/projects/${match.params.projectName}/jobs/${
+                    content?.find(item => item.uid === rowItem.uid.value)?.uid
```
Comment (mavdryk) [inline src/elements/JobsTableRow/JobsTableRow.js]: use _.find

File: src/utils/generateTableContent.js [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+
+export const generateTableContent = (
+  content,
+  groupedByName,
+  groupedByWorkflow,
+  groupFilter,
+  page
+) => {
+  if (!isEmpty(groupedByName) && groupFilter === 'name') {
+    return Object.values(groupedByName).map(group =>
```
Comment (mavdryk) [inline src/utils/generateTableContent.js]: you can get rid of `Object.values`

**Review by mavdryk — CHANGES_REQUESTED** [review]
